src/rt/test/perf/banking/verona_banks.cc

```diff
@@ -0,0 +1,113 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+#include "args.h"
+
+#include <cpp/when.h>
+#include <ds/scramble.h>
+#include <memory>
+#include <test/harness.h>
+
+/**
+ * This file implements an example of a set of Workers
+ * creating transactions.  The transactions use Accounts,
+ * and finally, all the transactions are logged.
+ *
+ * There is a busy loop in the transactions.
+ */
+
+struct Account
+{
+  int64_t balance;
+  int64_t overdraft;
+  size_t id;
+};
+
+using Accounts = std::vector<cown_ptr<Account>>;
+
+struct Worker
+{
+  std::shared_ptr<Accounts> accounts;
+
+  xoroshiro::p128r64 rand;
+
+  Worker(std::shared_ptr<Accounts> accounts, size_t seed) : accounts(accounts)
+  {
+    rand.set_state(seed + 1);
+  }
+};
+
+struct Log
+{};
+
+void log(cown_ptr<Log> log, std::string)
+{
+  when(log) << [=](auto) { /*std::cout << msg << std::endl;*/ };
+}
+
+void bank_job(acquired_cown<Worker>& worker, cown_ptr<Log> l, size_t repeats)
+{
+  // Select two accounts at random.
+  auto from_idx = worker->rand.next() % NUM_ACCOUNTS;
+  auto to_idx = from_idx;
+
+  // We don't want to use the same account.
+  while (to_idx == from_idx)
+    to_idx = worker->rand.next() % NUM_ACCOUNTS;
+
+  // Schedule a transfer
+  int64_t amount = 100;
+  Accounts& accounts = *worker->accounts;
+  when(accounts[from_idx], accounts[to_idx])
+    << [=](acquired_cown<Account> from, acquired_cown<Account> to) {

```

Comment:
Can this be:

```c++
    << [=](auto from, auto to) {
```

Not a suggested change, but if this works then it would be nice for a comment to saying that type inference works here and you can use `auto` to name the `cown`s in the lambda, but the types are given here explicitly to make the example clearer.